account_invoice_production_lot/models/account_invoice.py

```diff
@@ -31,10 +31,19 @@ class AccountInvoiceLine(models.Model):
         compute='_compute_line_lots',
     )
 
+    @api.multi
+    def _get_stock_customer_location_id(self):
+        self.ensure_one()
+        order = self.sale_line_ids[0].order_id
+        return order.partner_shipping_id.property_stock_customer
+
     @api.multi
     def _compute_prod_lots(self):
         for line in self:
-            line.prod_lot_ids = line.move_line_ids.mapped('lot_ids')
+            location_id = line._get_stock_customer_location_id()
+            quant_ids = line.move_line_ids.mapped('quant_ids').filtered(
```

Comment:
You should probably identify the delivered lots in a similar way to [how the `sale.order.line` computes the delivered qty](https://github.com/odoo/odoo/blob/0b414cb8145381cd8a3946aae9cec1601b111238/addons/sale_stock/models/sale_order.py#L176-L189).

Best way in my opinion is to start from `line.move_line_ids` (as you are already doing here).

Reply:
@SimoRubi did you mean https://github.com/tafaRU/account-invoice-reporting/commit/4697ce67ee33db1cb72844e7dd4ebc8932621101?
Tests are all passed :heavy_check_mark:

Reply:
yes @tafaRU that is the base idea, but the changes of https://github.com/tafaRU/account-invoice-reporting/commit/4697ce67ee33db1cb72844e7dd4ebc8932621101 are not included in this PR

Reply:
Ok thanks.

> but the changes of [tafaRU@4697ce6](https://github.com/tafaRU/account-invoice-reporting/commit/4697ce67ee33db1cb72844e7dd4ebc8932621101) are not included in this PR

Yep, I was waiting for your confirmation!

Reply:
@SimoRubi just force pushed. Please update your review.